# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/thebes/gfxPlatform.cpp
# Commit: 10b30c8d4b86
# Full Hash: 10b30c8d4b860b7d39de8d46fd186474407626b2
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2021-05-14 21:58:21
# Regressor Bug: 1708768
# File Overlap Count: 1
# Description:
#   Bug 1708768 - Create a pref to defer gfxPlatformFontList initialization until first use, and enable this on macOS. r=lsalzman
#   
#   This gives the RegisterFonts thread more time to complete its work, so that
#   the main thread doesn't have to wait for it in InitFontList.
#   
# ==============================================================================

diff -r c03c7df7cd61 -r 10b30c8d4b86 gfx/thebes/gfxPlatform.cpp
--- a/gfx/thebes/gfxPlatform.cpp	Fri May 14 18:05:44 2021 +0000
+++ b/gfx/thebes/gfxPlatform.cpp	Fri May 14 18:05:44 2021 +0000
@@ -968,8 +968,13 @@
 
   gPlatform->mHasVariationFontSupport = gPlatform->CheckVariationFontSupport();
 
-  if (!gPlatform->CreatePlatformFontList()) {
-    MOZ_CRASH("Could not initialize gfxPlatformFontList");
+  if (!StaticPrefs::gfx_font_list_lazy_init_enabled_AtStartup()) {
+    // On macOS we don't initialize the font list here because we want to allow
+    // more time for the RegisterFonts thread to complete. So it is initialized
+    // lazily on first use; see gfxPlatformFontList::PlatformFontList().
+    if (!gPlatform->CreatePlatformFontList()) {
+      MOZ_CRASH("Could not initialize gfxPlatformFontList");
+    }
   }
 
   gPlatform->mScreenReferenceSurface = gPlatform->CreateOffscreenSurface(